### PR TITLE
DocumentationExtraction: diff cut-off fix

### PR DESCRIPTION
**Pull request by damngamerz**
Fix diff's cut-off bug.
Fixes https://github.com/coala/coala/issues/4187

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by damngamerz** [comment]
There's no existing way to actually see this working unless DocStyleBear gets merged.
refer this https://hastebin.com/camuxuqebe.rb
CC @SanketDG @NiklasMM

**Comment by NiklasMM** [comment]
So as you noticed this breaks the tests. For example [this one](https://github.com/coala/coala/blob/master/tests/bearlib/languages/documentation/DocumentationExtractionTest.py#L165)
which has indentation in the docstring which is now lost due to your change.

I suggest using the smallest indentation of the whole block as the baseline.

File: tests/bearlib/languages/documentation/DocumentationExtractionTest.py [hunk 1]
```diff
@@ -180,8 +180,15 @@ def test_extract_documentation_PYTHON3(self):
...
-
+                        TextRange.from_values(40, 1, 42, 9)),
+                    DocumentationComment(
+                        ('\n'
+                         'A bad indented docstring\n'
+                         '    Improper indentation.\n'
+                         ':param impact: The force of Impact.\n'),
+                        docstyle_PYTHON3_default, '',
+                        docstyle_PYTHON3_default.markers[0],
+                        TextRange.from_values(45, 5, 49, 8)))
```
Comment (jayvdb) [inline tests/bearlib/languages/documentation/DocumentationExtractionTest.py]: Add a trailing `,` (and move the closing `)` to the next line, so you can add another one later without modifying this line.

**Comment by jayvdb** [comment]
ack e41d6b5

**Comment by jayvdb** [comment]
@rultor merge

**Pull request merged**